lib/v8_extras/CommonOperations.js

```diff
@@ -0,0 +1,334 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+//
```

Comment:
we get the entire stream implementation at pretty much zero startup cost by using v8 extras.

Reply:
Yes, I understand how `v8-extras` works, but that misses the point. The fact that it is bootstrapped into Node.js in a manner that is completely unlike every other thing in core means that we need to be deliberate about using this approach.

@hashseed ... what is the current state of support for `v8-extras`. I know there was discussion a few months ago about it being something that Node.js should *not* rely on but I don't know where that conversation ended up. Can you give us an update?

Reply:
I'd certainly like @hashseed's opinion on this, but I'd like to stress that this code is copied straight from Chromium. In other words, having it as a V8 Extra incurs no more maintenance burden on V8 Extras than "keep Chromium working".

Reply:
Keep in mind that `node-chakracore` is a thing that should also be able to support this. @nodejs/node-chakracore ... any thoughts on this being implemented as v8-extras?

Reply:
I've just had a quick scan through the code in v8_extras, and nothing looks too magical, although there are definitely v8-isms that node-chakracore will need to shim out which will take some work. Personally I'd prefer if it was less v8-ish and more node-ish, but I recognize that would require more work than grabbing chromium's implementation as was done here. My main concern at the moment is how this fits into the current build, since I really don't want to mess around with making sure our build process matches up with v8's in terms of random side effects like compiling js files into native.

One note on switching to using require: Right now all these v8_extras are written in library-style, where they take strong precautions to insulate themselves from environmental changes (e.g. no reference to the global Promise after initialization, since that may have been tainted by user code). Switching to a require style will change assumptions about when the code is run and lead to unexpected behavior. E.g. `var oldBool = Boolean; global.Boolean = myFakeBoolean; require("v8_extras"); global.Boolean = oldBool;` would inject a different class in the place of `Boolean` for the rest of this process, if that was the first require of v8_extras, and so the ordering of when packages are required could have unexpected impact.